### PR TITLE
systemd + apt/source.list+key

### PR DESCRIPTION
**Pull request by jo-so**
Systemd guesses the service is named *python*, because that's the
executable's name, but *matrix-synapse* would be better in the whole
journal.

### CODE REVIEW

File: debian/install [hunk 1]
```diff
@@ -1,2 +1,4 @@
 debian/homeserver.yaml etc/matrix-synapse
 debian/log.yaml etc/matrix-synapse
+debian/matrix-synapse.list etc/apt/sources.list.d
+debian/matrix-synapse.gpg etc/apt/trusted.gpg.d
```
Comment (uhoreg) [inline debian/install]: Ugh.  I hate it when packages stuff things in /etc/apt.  Please, no.

Comment (jo-so) [reply debian/install]: It's also done by *debian-ports-archive-keyring* and *debian-archive-keyring*. What's wrong with using these directories?

```
% dpkg -L debian-ports-archive-keyring debian-archive-keyring G etc/apt/
/etc/apt/trusted.gpg.d
/etc/apt/trusted.gpg.d/debian-ports-archive-2017.gpg
/etc/apt/trusted.gpg.d/debian-ports-archive-2018.gpg
/etc/apt/trusted.gpg.d/debian-ports-archive-2019.gpg
/etc/apt/trusted.gpg.d
/etc/apt/trusted.gpg.d/debian-archive-jessie-automatic.gpg
/etc/apt/trusted.gpg.d/debian-archive-jessie-security-automatic.gpg
/etc/apt/trusted.gpg.d/debian-archive-jessie-stable.gpg
/etc/apt/trusted.gpg.d/debian-archive-stretch-automatic.gpg
/etc/apt/trusted.gpg.d/debian-archive-stretch-security-automatic.gpg
/etc/apt/trusted.gpg.d/debian-archive-stretch-stable.gpg
/etc/apt/trusted.gpg.d/debian-archive-wheezy-automatic.gpg
/etc/apt/trusted.gpg.d/debian-archive-wheezy-stable.gpg
```

Comment (jo-so) [reply debian/install]: For reference: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=66325

**Comment by bombadil** [comment]
@jo-so, the difference with _debian-ports-archive-keyring_ and _debian-archive-keyring_ is that these are dedicated keyring packages, where this kind of action is the sole purpose of this package. Most users of this package (me included) definitely would not expect this from the synapse package.

The problem I assume you are trying to solve is better solved by getting the package into the official Debian archive.

**Comment by uhoreg** [comment]
> @jo-so, the difference with debian-ports-archive-keyring and debian-archive-keyring is that these are dedicated keyring packages, where this kind of action is the sole purpose of the package. Most users of this package (me included) definitely would not expect this from the synapse package.

Exactly.  There is a big difference between having a package whose purpose is to add trusted key to the keyring, versus having some random vendor package hijacking my system update functionality.  If you want to put those files in a separate package that's specifically for adding apt stuff, then that's one thing, but putting it in the main package will result in angering people.

> The problem I assume you are trying to solve is better solved by getting the package into the official Debian archive.

For one thing, matrix-synapse is already in Debian (unstable/testing).  But the Debian version will likely always lag behind the matrix.org version, especially when in comes to Debian stable.

**Comment by jo-so** [comment]
I know at least the third-party package for *Skype* does the same. This eases the installation/removal of such out-of-archive packages. You can download the package file and install it with `apt install ./file.deb` and after this *apt* cares about updates. When an upgrade to the package from the Debian archive happens, the reference to the third-party archive gets removed and the user doesn't have to care for dropping the sources.list and apt-key entries. I see this as an advantage in usability for third-party packages.

**Comment by uhoreg** [comment]
Google Chrome does the same thing, and believe me, I would have uninstalled it immediately after I found out what it had done, if my work at the time didn't require me to use it.  The fact is that it is unexpected behaviour for a package to add stuff to /etc/apt and I consider doing so to be quite rude.

Also, synapse is different from Skype and Chrome in that 1) it is also in Debian, and 2) the repository contains packages other than just matrix-synapse.  Consider the case where someone normally has synapse installed from the Debian repositories, but downloads the `.deb` from matrix.org for whatever reason (say to address a security issue, without wanting to wait for Debian to update), but still wanting to follow the Debian version.  The `.deb` would then cause their apt to follow the matrix.org repository (which they do not want to do), and may result in other packages to be unexpectedly "upgraded" from the matrix.org repository (which again may be unwanted).

Also, you've set the sources.list to use the stretch repository, which is obviously not correct for non-stretch users.

**Comment by uhoreg** [comment]
Also, the systemd and the sources.list/apt-key changes really should be two different PRs, especially given that you've already done it as two separate commits.

**Comment by jo-so** [comment]
There's a better patch for systemd integration on https://salsa.debian.org/matrix-team/matrix-synapse/commit/ff086b91c0c7a34ab2167178174cbe3978bd442e Maybe you can pick this.

**Pull request closed without merging**